### PR TITLE
[MBL-15977][Student][Teacher] Discussion redesign

### PR DESCRIPTION
**Pull request by hermannakos**
refs: MBL-15977
affects: Student, Teacher
release note: none

test plan: Enable the discussion redesign feature flag for your sandbox (you can enable/disable it by course as well). Smoke test around the Discussion, Announcements and Group Discussions.

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/discussion/details/DiscussionDetailsWebViewViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    private val _state = MutableLiveData<ViewState>()
+
+    fun loadData(canvasContext: CanvasContext, discussionTopicHeader: DiscussionTopicHeader) {
+        viewModelScope.launch {
+            try {
+                _state.postValue(ViewState.Loading)
+                val locale = Locale.getDefault().language
+                val timezone = TimeZone.getDefault().id
+                val url = "${apiPrefs.fullDomain}/${canvasContext.apiContext()}/${canvasContext.id}/discussion_topics/${discussionTopicHeader.id}"
+                val authenticatedUrl = "${oauthManager.getAuthenticatedSessionAsync(url).await().dataOrThrow.sessionUrl}&embed=true&session_locale=$locale&session_timezone=$timezone"
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/features/discussion/details/DiscussionDetailsWebViewViewModel.kt:59]: It looks a bit weird to embed an async call into a string.

File: libs/pandautils/src/main/res/layout/fragment_discussion_details_web_view.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+
+    <data>
+
+        <variable
+            name="viewModel"
+            type="com.instructure.pandautils.features.discussion.details.DiscussionDetailsWebViewViewModel" />
+    </data>
+
+    <FrameLayout
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/res/layout/fragment_discussion_details_web_view.xml:26]: Do we need this FrameLayout? I think the `EmptyView` can be in the `ConstraintLayout` as well.

**Review by tamaskozmer — CHANGES_REQUESTED** [review]

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by hermannakos** [comment]
refs: MBL-15977
affects: Student, Teacher
release note: none

test plan: Enable the discussion redesign feature flag for your sandbox (you can enable/disable it by course as well). Smoke test around the Discussion, Announcements and Group Discussions.

**Pull request merged**